### DISCUSSION TITLE
Help on install on Armbian Bookworm.

### DISCUSSION THREAD

**Discussion by lamerjack** [bimmerconnected/bimmer_connected#572] [Q&A]
Hi, sorry i have no big experience with python so i need an help to install after os upgrade.
With Armbian Bullseye no problem i wrote sudo pip install bimmer_connected and everything was fine:
all user can run bimmerconnected from shell and i was able to use in simple python scripts.

In Bookworm:
sudo pip3 install bimmer_connected

    error: externally-managed-environment

    This environment is externally managed
    To install Python packages system-wide, try apt install
    python3-xyz, where xyz is the package you are trying to
    install.

    If you wish to install a non-Debian-packaged Python package,
    create a virtual environment using python3 -m venv path/to/venv.
    Then use path/to/venv/bin/python and path/to/venv/bin/pip. Make
    sure you have python3-full installed.

    If you wish to install a non-Debian packaged Python application,
    it may be easiest to use pipx install xyz, which will manage a
    virtual environment for you. Make sure you have pipx installed.

with apt install of course doesn't work.
so i tryed with 
sudo pipx install bimmer_connected
sudo pipx ensurepath install bimmer_connected

but i have not bimmerconnected in the path.
probably is some basic usage of the pipx installer... but as i already said my knowledge of python is very low
Thanks in advance for your help.

**Comment by rikroe** [comment]
Apparently, bullseye doesn't allow packages to be installed in the system python environment (like C:\program files in windows).

That is an OS decision, not related to the library or how you're installing it. 

You can read up on "python virtual environment" via Google (that's the long term for venv).

An alternative could be a pip `--user` install.
